lib/Doctrine/Migrations/Version/Executor.php

```diff
@@ -183,161 +171,162 @@ private function startMigration(
     {
-        $versionInfo = $version->getVersion();
+        $versionInfo = (string) $planItem->getVersion();
         $description = $migration->getDescription();
 
         if ($description !== '') {
             $versionInfo .= ' (' . $description . ')';
         }
 
+        $params = ['version_name' => $versionInfo];
+
         if ($direction === Direction::UP) {
-            return sprintf('  <info>++</info> migrating <comment>%s</comment>', $versionInfo);
+            return ['++ migrating {version_name}', $params];
         }
 
-        return sprintf('  <info>--</info> reverting <comment>%s</comment>', $versionInfo);
+        return ['++ reverting {version_name}', $params];
     }
 
-    private function skipMigration(
-        SkipMigration $e,
-        Version $version,
-        AbstractMigration $migration,
-        string $direction,
-        MigratorConfiguration $migratorConfiguration
-    ) : void {
+    private function migrationEnd(Throwable $e, MigrationPlan $plan, ExecutionResult $result, MigratorConfiguration $configuration) : void
+    {
+        $plan->setResult($result);
+        $this->logResult($e, $result, $plan);
+
+        $this->dispatcher->dispatchVersionEvent(
+            Events::onMigrationsVersionSkipped,
+            $plan,
+            $configuration
+        );
+
+        $migration = $plan->getMigration();
+
         if ($migration->isTransactional()) {
             //only rollback transaction if in transactional mode
             $this->connection->rollBack();
         }
 
-        if (! $migratorConfiguration->isDryRun()) {
-            $version->markVersion($direction);
+        if ($configuration->isDryRun() || $result->isSkipped() || $result->hasError()) {
+            return;
         }
 
-        $this->outputWriter->write(sprintf("\n  <info>SS</info> skipped (Reason: %s)", $e->getMessage()));
-
-        $version->setState(State::NONE);
-
-        $this->configuration->dispatchVersionEvent(
-            $version,
-            Events::onMigrationsVersionSkipped,
-            $direction,
-            $migratorConfiguration->isDryRun()
-        );
+        $this->metadataStorage->complete($result);
     }
 
-    /**
-     * @throws Throwable
-     */
-    private function migrationError(Throwable $e, Version $version, AbstractMigration $migration) : void
+    private function logResult(Throwable $e, ExecutionResult $result, MigrationPlan $plan) : void
     {
-        $this->outputWriter->write(sprintf(
-            '<error>Migration %s failed during %s. Error %s</error>',
-            $version->getVersion(),
-            $version->getExecutionState(),
-            $e->getMessage()
-        ));
-
-        if ($migration->isTransactional()) {
-            //only rollback transaction if in transactional mode
-            $this->connection->rollBack();
+        if ($result->isSkipped()) {
+            $this->logger->error(
+                'Migration {version} skipped during %s. Reason {error}',
```

Comment:
Shouldn't the %s be replaced with `{state}` here and below?

Reply:
good catch, fixed